src/metabase/search/filter.clj

```diff
@@ -117,8 +125,10 @@
   "Return a HoneySQL clause corresponding to all the optional search filters."
   [search-context qry]
   (as-> qry qry
-    (sql.helpers/where qry (when (seq (:models search-context))
-                             [:in :search_index.model (:models search-context)]))
+    (sql.helpers/where qry (if (seq (:models search-context))
```

Comment:
We handle defaulting to "all models" if the parameter is missing when we build the search context.

Reply:
do we? `search-context` complains if you don't provide `:models`, and just returns empty set if you give it nil.

Reply:
Yeah that's what I'm thinking. I'll do it in another PR because I want to backport these fixes ASAP and worried there might be a lot of tests to update.